src/Quotebot.Data/CosmosDbService.cs

```diff
@@ -28,85 +29,188 @@ public async Task<bool> TryCreateQuoteRecord(Quoted message)
+        //}
+
         public async Task<string> FindByQuoteInServer(string messageLike, int take = 5)
         {
-            var iterator = await _container.GetItemQueryIterator<Quoted>().ReadNextAsync();
-            if (!iterator.Any())
+            var results = await FindQuotesInServer(messageLike, take);
+
+            var enumerable = results as Quoted[] ?? results.ToArray();
+            if (!enumerable.Any())
             {
                 return $"No quotes found containing the text *{messageLike}* in this server.";
             }
 
-            using var setIterator = _container.GetItemLinqQueryable<Quoted>(allowSynchronousQueryExecution: true)
-                                 .Where(record => record.CleanContent != null && record.CleanContent.Contains(messageLike, StringComparison.InvariantCultureIgnoreCase))
-                                 .Take(take)
-                                 .ToFeedIterator();
+            StringBuilder stringBuilder = new();
+
+            foreach (var quote in enumerable)
+            {
+                stringBuilder
+                    .AppendLine()
+                    .AppendLine(
+                        $"{quote.CreatedAt.ToString("d")} - **{quote.Author.Nickname ?? quote.Author.Username}** in #{quote.Channel.Name} : {quote.Content}");
+            }
+
+            var resultString = stringBuilder.ToString();
+
+            return resultString;
+        }
+
+        public async Task<string> FindByQuote(string messageLike, string channelName, int take = 5)
+        {
+            IEnumerable<Quoted> results = await FindQuotesByChannel(messageLike, channelName, take);
+            var quotesList = results.ToList();
+            if (!quotesList.Any())
+                return $"No quotes found containing the text *{messageLike}* in this channel.";
 
-            List<Quoted> results = new();
 
             StringBuilder stringBuilder = new();
-            while (setIterator.HasMoreResults)
+            foreach (var quote in quotesList)
             {
-                foreach (var quote in await setIterator.ReadNextAsync())
-                {
-                    stringBuilder
-                        .AppendLine()
-                        .AppendLine($"{quote.CreatedAt.ToString("d")} - **{quote.Author?.Nickname ?? quote.Author?.Username}** in #{quote.Channel.Name} : {quote.Content}");
-                }
+                stringBuilder
+                    .AppendLine()
+                    .AppendLine(
+                        $"{quote.CreatedAt.ToString("d")} - **{quote.Author.Nickname ?? quote.Author.Username}** : {quote.Content}");
             }
 
             var result = stringBuilder.ToString();
-            if(string.IsNullOrWhiteSpace(result))
-                return $"No quotes found containing the text *{messageLike}* in this server.";
+            if (string.IsNullOrWhiteSpace(result))
+                return $"No quotes found containing the text *{messageLike}* in this channel.";
```

Comment:
Is it useful to emit channelName here since you have it?

Reply:
Yeah, probably